subsys/net/lib/lwm2m/lwm2m_obj_device.c

```diff
@@ -294,12 +372,25 @@ static int lwm2m_device_init(void)
 		LOG_DBG("Create LWM2M instance 0 error: %d", ret);
 	}
 
-	/* Create the default error code resource instance */
-	lwm2m_device_add_err(0);
+	/* Ensure error list is reset if not loaded from settings */
+	reset_error_list();
+
+	/* Load error code resource instances */
+#ifdef CONFIG_LWM2M_DEVICE_ERROR_CODE_SETTINGS
```

Comment:
```suggestion
if (IS_ENABLED(CONFIG_LWM2M_DEVICE_ERROR_CODE_SETTINGS) {
```

I believe that if you use `IS_ENABLED()` macro, you can write this whole feature without using `#ifdef`.
Compiler and linker will then drop away the parts that are not used and not reachable.

Reply:
Good idea, done